### PR TITLE
CHORE: Add conventional commits steps

### PR DESCRIPTION
**Pull request by alessio-libardi-zupit**

### CODE REVIEW

File: .github/workflows/conventional-commits-step-lint.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: wagoid/commitlint-github-action@v5
+        with:
+          configFile: ${{ inputs.CONFIG_FILE }}
+
+
+  lint-pr:
+    name: Lint PR
```
Comment (ivan-morandi-zupit) [inline .github/workflows/conventional-commits-step-lint.yml:28]: qua prende in automatico il branch / codice? visto che prima hai fatto il checkout ma non qua

File: README.md [hunk 2]
```diff
@@ -80,36 +86,40 @@ For example:
...
 
 ###### Action
+
 **.github/actions/node/build** is the action that builds a NodeJS project.
 
 It requires these inputs:
+
 - **NODE_VERSION**: The NodeJS version required to build the project.
-- **RELEASE_ENVIRONMENT**: The environment for which the project must be compiled (e.g. *testing*, *staging*, *production*).
+- **RELEASE_ENVIRONMENT**: The environment for which the project must be compiled (e.g. _testing_, _staging_, _production_).
```
Comment (ivan-morandi-zupit) [inline README.md:110]: è voluto questo cambio? (e anche nel resto del readme)

Comment (alessio-libardi-zupit) [reply README.md:110]: Oh no, l'editor ha fatto molto più del dovuto oggi 😢

**Comment by alessio-libardi-zupit** [comment]
Ho aggiunto un `.editorconfig` in modo tale che ci sia uno standard che gli IDE rispettino.

Per presa visione di seguito la sezione aggiunta al `README.md`:

### Conventional Commits

#### Conventional Commits Lint Step

**conventional-commits-step-lint.yml** is the workflow that lint the commit messages of a pull request.

It requires these inputs:

-   **LABELS**: the _labels_ to select the correct _github-runner_ that will execute this workflow. The format is a stringified JSON list of labels.
-   **CONFIG_FILE**: the config file name (by default it is **.commitlintrc**).

This is an example to show how data should be formatted.

```yaml
jobs:
    lint-pr:
        uses: zupit-it/pipeline-templates/.github/workflows/conventional-commits-step-lint.yml@main
        with:
            LABELS: "['pinga', 'pipeline', 'native']"
            CONFIG_FILE: .commitlintrc.json
        secrets: inherit
```

#### Conventional Commits Release Step

**conventional-commits-step-release.yml** is the workflow that automatically creates a new release based on the commit messages.

It requires these inputs:

-   **LABELS**: the _labels_ to select the correct _github-runner_ that will execute this workflow. The format is a stringified JSON list of labels.

It also requires these secrets:

-   **RELEASE_TOKEN**: a personal access token with grants to create a release and to push new commits. (use the zupit bot)

This is an example to show how data should be formatted.

```yaml
jobs:
    lint-pr:
        uses: zupit-it/pipeline-templates/.github/workflows/conventional-commits-step-release.yml@main
        with:
            LABELS: "['pinga', 'pipeline', 'native']"
        secrets: inherit
```

**Review by alessandro-mariotti-zupit — APPROVED** [review]

**Pull request merged**